scifight/utils.py

```diff
@@ -0,0 +1,32 @@
+
+def shorten_text(text, maxchars, ellipsis='...'):
+    """ Returns first few words from the `text`, but no more than `maxchars`, and appends
+        an `ellipsis`. This function works only on plain text (no HTML!), removes any leading
+        and trailing spaces, and squashes any number of consequent whitespaces (including
+        newlines) into a single space.
+
+        :param text: Input string, possibly quite long.
+        :param maxchars: Maximum length or result string.
+        :param ellipsis: Trailing sign to indicate implied continuation.
+        :return: Shortened substring with squashed whitespaces.
+    """
+    text = text.lstrip().rstrip().replace('\n', '')
```

Comment:
Nice try, but…
* Aren't `lstrip().rstrip()` equivalent to just `strip()`?
* Replacing newlines with empty string may suddenly merge words together.
* If your text is extremely large (say, 10 megabytes long) you're going to take at least three copies of it.